pkg/segment/structs/evaluationstructs.go

```diff
@@ -1285,6 +1285,60 @@ func (self *NumericExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure)
 			return math.Abs(left), nil
 		case "ceil":
 			return math.Ceil(left), nil
+		case "acosh":
+			if left < 1 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: acosh requires values >= 1, got: %v", left)
+			}
+			return math.Acosh(left), nil
+		case "acos":
+			if left < -1 || left > 1 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: acos requires values between -1 and 1, got: %v", left)
+			}
+			return math.Acos(left), nil
+		case "asin":
+			if left < -1 || left > 1 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: asin requires values between -1 and 1, got: %v", left)
+			}
+			return math.Asin(left), nil
+		case "asinh":
+			return math.Asinh(left), nil
+		case "atan":
+			return math.Atan(left), nil
+		case "atanh":
+			if left <= -1 || left >= 1 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: atanh requires values between -1 and 1 exclusive, got: %v", left)
+			}
+			return math.Atanh(left), nil
+		case "cos":
+			return math.Cos(left), nil
+		case "cosh":
+			return math.Cosh(left), nil
+		case "sin":
+			return math.Sin(left), nil
+		case "sinh":
+			return math.Sinh(left), nil
+		case "tan":
+			// Check for points where cos(x) = 0, which would cause tan to be undefined.
+			// These are points (pi/2 + k*pi) where k is an integer.
+			// To check for this, see if left modulo pi is pi/2 (or very close due to floating point precision).
+			halfPi := math.Pi / 2
+			mod := math.Mod(left, math.Pi)
+			if math.Abs(mod-halfPi) < 0.0000001 || math.Abs(mod+halfPi) < 0.0000001 {
+				return -1, fmt.Errorf("NumericExpr.Evaluate: tan is undefined at pi/2 + k*pi, got: %v", left)
+			}
+			return math.Tan(left), nil
```

Comment:
I don't think the error handling will work for some negative values. For example `-3pi/2`.

From https://pkg.go.dev/math#Mod
> Mod returns the floating-point remainder of x/y. The magnitude of the result is less than y and its sign agrees with that of x.

So your `mod` would be `-pi/2` and `mod-halfPi` would be `-pi` instead of `0`

Reply:
In cases where we have a negative value like `-3pi/2`, `math.Abs(mod+halfPi)` will be very close to zero, so the condition will be true, and the function will correctly return an error. Am I missing something here?

Reply:
`math.Mod` sometimes returns negative numbers. https://go.dev/play/p/X2ZNWbyOemt